File: Array/Array/ViewController.m
//
//  ViewController.m
//  Array
//
//  Created by AaronYin on 2020/11/15.
//

#import "ViewController.h"
#import "ZZArray.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
//    NSUInteger i = 0;
//    NSLog(@"%lu", i++);
//    NSLog(@"%lu", i);
//    NSLog(@"%lu", ++i);
//    NSLog(@"%lu", i);
    
    ZZArray *array = [[ZZArray alloc] init];
    
    [array addObject:@"zzz"];
    
    for (NSInteger i = 1; i < 11; i++) {
        [array addObject:[NSString stringWithFormat:@"%ld", i]];
    }
    
    [array removeObjectAtIndex:4];
//    [array removeAllObjects];
    [array removeLastObject];
    
    [array insertObject:@"999" atIndex:8];
    
    NSLog(@"%@", array);
    
    [array release];
    
}


@end

File: BigO/BigO/ViewController.swift
//
//  ViewController.swift
//  BigO
//
//  Created by AaronYin on 2020/11/14.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        
//        print(fib2(6))
//        print(fib2(7))
//        print(fib2(8))
        
        performanceMeasure {
            _ = fib1(45)
        }
        
        performanceMeasure {
            _ = fib2(45)
        }
        
    }


}

